### PR TITLE
Adding a run-action to upgrade Docker

### PR DESCRIPTION
**Pull request by joedborg**
~~**DON'T MERGE BEFORE DISCUSSION.**~~

I've moved some functions around so they can be shared with the run action easily, this works with just deploying the layer, but we should test with some consumers (i.e. kubernetes-worker).

### CODE REVIEW

**Comment by joedborg** [comment]
[docker upgrades on workers](https://trello.com/c/gvz9DP1O/1171-docker-upgrades-on-workers)

File: lib/charms/layer/docker.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            docker_runtime = 'nvidia'
+        else:
+            docker_runtime = 'apt'
+
+    hookenv.log(
+        'Setting runtime to {}'.format(docker_packages))
+    return docker_runtime
+
+
+def render_configuration_template(service=False):
```
Comment (hyperbolic2346) [inline lib/charms/layer/docker.py]: Do we anticipate ever rendering the configuration without rendering the service? It seems everything calls this with service=True.

**Comment by joedborg** [comment]
Have just built `kubernetes-worker` using this layer and it's working fine.

File: lib/charms/layer/docker.py (hunk 1 above)
Comment (joedborg) [reply lib/charms/layer/docker.py]: I don't think this one is necessary https://github.com/juju-solutions/layer-docker/pull/135/files/0a6b0f97fa9446d0ae21dc872c2fe3d1a783fd29#diff-b72ba34de38636dd3d255bd7c29c1e73R736

But it was the old behavior so wanted to check before I changed it.

**Comment by tvansteenburgh** [comment]
Gonna need docs for this.

/cc @evilnick

**Comment by joedborg** [comment]
https://github.com/juju-solutions/kubernetes-docs/pull/106

**Pull request merged**